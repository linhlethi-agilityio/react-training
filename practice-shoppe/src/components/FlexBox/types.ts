import { TUnitValue } from '@type/unitValue';
import { TStyleUnit } from '@helper/transformNumber';

export type TJustifyContent = 'center' | 'space-between' | 'left' | 'right';

export interface IFlexBoxProps {
  gap?: TUnitValue;
  unitFlexBox?: TStyleUnit;
  justifyContent?: TJustifyContent;
  mTop?: TUnitValue;
  mLeft?: TUnitValue;
  mRight?: TUnitValue;
  mBottom?: TUnitValue;
}
