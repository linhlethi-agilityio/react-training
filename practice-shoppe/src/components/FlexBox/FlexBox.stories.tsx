import { ComponentMeta, ComponentStory } from '@storybook/react';
import { StyledFlexBox } from '@components/FlexBox';

export default {
  title: 'FlexBox',
  component: StyledFlexBox,
} as ComponentMeta<typeof StyledFlexBox>;

const Template: ComponentStory<typeof StyledFlexBox> = (args) => (
  <StyledFlexBox {...args}>
    <div>Flex 1</div>
    <div>Flex 2</div>
  </StyledFlexBox>
);

export const CenterFlexBox = Template.bind({});
const centerFlexBox = {
  gap: 10,
  justifyContent: 'center',
};
CenterFlexBox.args = centerFlexBox;

export const SpaceBetweenFlexBox = Template.bind({});
const spaceBetweenFlexBox = {
  gap: 10,
  justifyContent: 'space-between',
};
SpaceBetweenFlexBox.args = spaceBetweenFlexBox;

export const RightFlexBox = Template.bind({});
const rightFlexBox = {
  gap: 10,
  justifyContent: 'right',
};
RightFlexBox.args = rightFlexBox;

export const LeftFlexBox = Template.bind({});
const leftFlexBox = {
  gap: 10,
  justifyContent: 'left',
};
LeftFlexBox.args = leftFlexBox;
