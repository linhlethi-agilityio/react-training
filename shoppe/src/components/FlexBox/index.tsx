import { memo } from 'react';
import { transformToStyleValue } from '@helper/transformNumber';
import styled from 'styled-components';
import { IFlexBoxProps } from './types';

const StyledFlexBox = memo(styled.div<IFlexBoxProps>`
  display: flex;
  gap: ${(props) => transformToStyleValue(props.gap, props.unitFlexBox)};
  justify-content: ${(props) => props.justifyContent};
  margin-left: ${(props) => transformToStyleValue(props.mLeft, props.unitFlexBox)};
  margin-right: ${(props) => transformToStyleValue(props.mRight, props.unitFlexBox)};
  margin-top: ${(props) => transformToStyleValue(props.mTop, props.unitFlexBox)};
  margin-bottom: ${(props) => transformToStyleValue(props.mBottom, props.unitFlexBox)};
`);

export { StyledFlexBox };
