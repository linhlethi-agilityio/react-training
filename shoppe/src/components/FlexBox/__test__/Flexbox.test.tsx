import { RenderResult, render } from '@testing-library/react';
import { StyledFlexBox } from '@components/FlexBox';
describe('testing Flexbox component', () => {
  let wrapper: RenderResult;
  beforeEach(() => {
    wrapper = render(<StyledFlexBox />);
  });
  it('matches snapshot', () => {
    const { asFragment } = wrapper;
    expect(asFragment()).toMatchSnapshot();
  });
});
