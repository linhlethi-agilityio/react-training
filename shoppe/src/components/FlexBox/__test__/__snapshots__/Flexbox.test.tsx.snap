// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Flexbox component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-beqWaB fDdkWj"
  />
</DocumentFragment>
`;
